### PR TITLE
Refactor oauth flow

### PR DESCRIPTION
**Pull request by NeonDaniel**
Removes redundant `self.bus` definition
Refactor oauth message handling to use `get_response` instead of registering handlers
Remove redundant `data=None` from `message.forward` calls causing editor warnings
Add annotations to methods

### CODE REVIEW

**Comment by NeonDaniel** [comment]
Needs https://github.com/OpenVoiceOS/ovos-PHAL-plugin-oauth/pull/15

File: ovos_PHAL_plugin_homeassistant/__init__.py [hunk 1]
```diff
@@ -752,55 +751,103 @@ def handle_set_group_display_settings(self, message):
...
     def handle_start_oauth_flow(self, message):
-        """ Handle the start oauth flow message
+        """
+        Handle a request to start oauth login, i.e. from a GUI page
 
             Args:
                 message (Message): The message object
         """
-        instance = message.data.get("instance", None)
+        instance = message.data.get("instance").lower()
```
Comment (mikejgray) [inline ovos_PHAL_plugin_homeassistant/__init__.py:790]: `None` will not have method `lower()`, so we'll need a default empty string. That will still evaluate as falsey in the line below and won't potentially throw an exception.

File: ovos_PHAL_plugin_homeassistant/__init__.py [hunk 2]
```diff
@@ -110,9 +110,8 @@ def __init__(self, bus=None, config=None):
         self.bus.on("configuration.patch", self.init_configuration)
 
         # LISTEN FOR OAUTH RESPONSE
-        self.bus.on("oauth.app.host.info.response", self.handle_oauth_host_info)
-        self.bus.on("oauth.generate.qr.response", self.handle_qr_oauth_response)
```
Comment (mikejgray) [inline ovos_PHAL_plugin_homeassistant/__init__.py:114]: Did you mean to remove these? Is that because the OAuth plugin handles them?

**Review by mikejgray — CHANGES_REQUESTED** [review]
Some nits and a potential type error, but nothing that should hold up testing. I'll load this up on my devkit and see how it goes

File: ovos_PHAL_plugin_homeassistant/__init__.py [hunk 3]
```diff
@@ -752,55 +751,103 @@ def handle_set_group_display_settings(self, message):
...
+            raise RuntimeError(resp.data['error'])
+        self.handle_qr_oauth_response(resp)
 
     def handle_qr_oauth_response(self, message):
-        qr_code_url = message.data.get("qr", None)
+        """
+        Handle a response message with a path to a QR code to display
+        @param message: oauth.generate.qr.response
+        """
+        qr_code_url = message.data.get("qr")
```
[the remark this answers is not in the document]
Comment (NeonDaniel) [reply ovos_PHAL_plugin_homeassistant/__init__.py:850]: Nothing is displayed; it probably would be worth raising an exception if we handled those exceptions and present them to the user somehow

File: ovos_PHAL_plugin_homeassistant/__init__.py (hunk 2 above)
Comment (NeonDaniel) [reply ovos_PHAL_plugin_homeassistant/__init__.py:114]: Yeah, these were refactored so `wait_for_response` will handle these during the flow in one thread. Couple reasons I opted for this:
- prevents handling events if another service is performing oauth
- allows handling one oauth flow in one thread so raised exceptions get to the original method
- allows raising timeout errors/knowing when an expected response didn't come

**Review by mikejgray — COMMENTED** [review]
Looks good - loading on a Mark 2 for testing, and if that goes well I'll approve

**Review by mikejgray — APPROVED** [review]

**Pull request merged**